lib/coverband/memory_cache_store.rb

```diff
@@ -32,12 +32,7 @@ def filter(files)
         #first time we see a file, we pre-init the in memory cache to whatever is in store(redis)
         line_cache = files_cache[file] ||= Set.new(store.covered_lines_for_file(file))
         lines.reject! do |line|
-          if line_cache.include? line
-            true
-          else
-            line_cache << line
-            false
-          end
+          line_cache.include?(line) ? true : (line_cache << line and false)
```

Comment:
I don't believe this is the case, I mean, method call inside blocks normally is **very expensive**, this "sacrifice" can be acceptable in cases where the domain logic is too heavy or complex, see this benchmarks:

```ruby
require 'benchmark'

arry = (0..100_000).to_a
interations = (ENV['ITERATIONS'] || 1000).to_i

def reject_filter(value)
  return value > 50_000
end

Benchmark.bm(50) do |bm|
  bm.report('reject with method call inside block') do
    interations.times do |i|
      arry.reject do |value|
        reject_filter(value)
      end
    end
  end

  bm.report('reject without method call inside block') do
    interations.times do |i|
      arry.reject do |value|
        value > 50_000
      end
    end
  end

  bm.report('reject with method to proc call') do
    interations.times {
      arry.reject(&method(:reject_filter))
    }
  end
end
```

My results:

```
➔ ITERATIONS=1000 ruby test.rb
                                                         user     system      total        real
reject with method call inside block                 8.720000   0.030000   8.750000 (  8.754245)
reject without method call inside block              5.760000   0.040000   5.800000 (  5.801007)
reject with method to proc call                     14.930000   0.000000  14.930000 ( 14.928754)
```